src/App/Fossa/VSI/DynLinked/Internal/Lookup.hs

```diff
@@ -31,13 +36,25 @@ dynamicDependencies ::
   -- Resulting dependencies.
   m (Set DynamicDependency)
 dynamicDependencies root files = do
-  apkLookupTable <- buildLookupTable root
-  resolved <- traverse (resolveFile apkLookupTable root) $ Set.toList files
+  apkLookupTable <- context "build APK lookup table" $ buildLookupTable root
+  let targets = Set.toList files
+  logDebug . pretty $ "Resolving files: " <> show targets
+  resolved <- traverse (resolveFile apkLookupTable root) targets
   pure $ Set.fromList resolved
 
-resolveFile :: (Has Diagnostics sig m, Has Exec sig m) => Maybe APKLookupTable -> Path Abs Dir -> Path Abs File -> m DynamicDependency
+resolveFile ::
+  ( Has Diagnostics sig m
+  , Has Exec sig m
+  , Has Logger sig m
+  ) =>
+  Maybe APKLookupTable ->
+  Path Abs Dir ->
+  Path Abs File ->
+  m DynamicDependency
 resolveFile table root file = do
-  resolved <- rpmTactic root file <||> debTactic root file <||> pure (apkTactic table file)
+  -- When adding new tactics in the future, ensure that they fail (through diagnostics) if they're not the last link in the chain.
+  -- <||> selects the first item to succeed without a diagnostic error.
+  resolved <- rpmTactic root file <||> debTactic root file <||> apkTactic table file
```

Comment:
I think they should just always fail.  That way, ordering is not critical for success.

```suggestion
  -- When adding new tactics in the future, ensure that they fail through diagnostics.
  -- <||> selects the first item to succeed without a diagnostic error.
  resolved <- rpmTactic root file <||> debTactic root file <||> apkTactic table file
```

Reply:
Idea for a utility: 

```haskell
dsum [a, b, c] = a <||> b <||> c
```

Naming comes from [`asum`](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Foldable.html#v:asum), but we should probably find a better name.

I think it could be implemented as 
```haskell
data DSumStart = DSumStart

instance ToDiagnostic DSumStart where
  renderDiagnostic _ = "dsum was called with an empty list"

dsum = foldr (<||>) (fatal DSumStart)
```